server/log_operation_manager.go

```diff
@@ -247,6 +251,42 @@ func (l *LogOperationManager) getLogIDs(ctx context.Context) ([]int64, error) {
 	return logIDs, nil
 }
 
+// logName maps a logID to a human-readable name, caching results along the way.
+// The human-readable name may non-unique so should only be used for diagnostics.
+func (l *LogOperationManager) logName(ctx context.Context, logID int64) string {
+	l.nameMutex.Lock()
+	defer l.nameMutex.Unlock()
+	if name, ok := l.logNames[logID]; ok {
+		return name
+	}
+	tx, err := l.info.Registry.AdminStorage.Snapshot(ctx)
+	if err != nil {
+		glog.Errorf("%v: failed to start transaction: %v", logID, err)
+		return "<err>"
+	}
+	defer tx.Close()
+	tree, err := tx.GetTree(ctx, logID)
+	if err != nil {
+		glog.Errorf("%v: failed to get log info: %v", logID, err)
+		return "<err>"
+	}
+	tx.Commit()
+	name := tree.DisplayName
```

Comment:
Suggestion: Always use `name := fmt.Sprintf("%v/%v", tree.DisplayName, tree.TreeId)`. This sorts out our problems with empty / duplicate names. If we can't read the name we could use the same format, but with "\<err\>/$id" or maybe just "/$id".

Reply:
I'm aiming at a display name that's for human readability, and assumes a sane configuration (all trees have `DisplayName`, no dups).  So I think that:
```
master for: argon2017 argon2020 fred bob
```
is vastly better than:
```
master for: argon2017/1736462159951049615 argon2020/7432457761086508362 fred/8590506478028889672 bob/172864892775937963
```